src/System.CommandLine/Option.cs

```diff
@@ -13,14 +15,43 @@ public class Option : Symbol, IOption
             Argument argument = null,
             bool isHidden = false)
             : base(aliases, description, argument, isHidden)
-        { }
+        {
+        }
 
         public Option(
             string alias,
             string description = null,
             Argument argument = null,
             bool isHidden = false)
-            : base(new [] {alias}, description, argument, isHidden)
-        { }
+            : base(new[]
+            {
+                alias
+            }, description, argument, isHidden)
+        {
+        }
+
+        public virtual Argument Argument
+        {
+            get => _arguments.FirstOrDefault() ?? Argument.None;
+            set
+            {
+                if (_arguments.Any())
+                {
+                    _arguments.Clear();
+                }
+
+                AddArgumentInner(value);
+            }
+        }
+
+        IArgument IOption.Argument => Argument;
+
+        string IValueDescriptor.ValueName => Name;
+
+        Type IValueDescriptor.Type => _arguments.Single().ArgumentType;
```

Comment:
Should this be `=> Argument.ArgumentType;`?
The same for `HasDefaultValue` and `GetDefaultValue()`. 

Since the argument is optional when instantiating the class, I suspect we should not assume the `_arguments` collection has any items.